filters/encrypt/filter.go

```diff
@@ -0,0 +1,554 @@
+	// check for nil value (prevent panics)
+	if v == reflect.ValueOf(nil) {
+		return nil
+	}
+
+	for i := 0; i < v.Type().NumField(); i++ {
+		field := v.Field(i)
+
+		switch v.Field(i).Kind() {
+		case reflect.Ptr, reflect.Interface:
+			field = v.Field(i).Elem()
+			if field == reflect.ValueOf(nil) {
+				continue
+			}
+		}
+
+		fkind := field.Kind()
+		ftype := field.Type()
+
+		var taggedInterface Taggable
+		var isTaggable bool
+		// check exported fields to see if they implement the Taggable interface
+		if field.CanSet() {
+			taggedInterface, isTaggable = v.Field(i).Interface().(Taggable)
+		}
+
+		switch {
+		// if the field is a string or []byte then we just need to sanitize it
+		case ftype == reflect.TypeOf("") || ftype == reflect.TypeOf([]uint8{}):
+			classificationTag := getClassificationFromTag(v.Type().Field(i).Tag, withFilterOperations(filterOverrides))
+			if err := ef.filterValue(ctx, field, classificationTag, opt...); err != nil {
+				return fmt.Errorf("%s: %w", op, err)
+			}
+		// if the field is a slice
+		case fkind == reflect.Slice:
+			switch {
+			// if the field is a slice of string or slice of []byte
+			case ftype == reflect.TypeOf([]string{}) || ftype == reflect.TypeOf([][]uint8{}):
+				classificationTag := getClassificationFromTag(v.Type().Field(i).Tag, withFilterOperations(filterOverrides))
+				if err := ef.filterSlice(ctx, classificationTag, field, opt...); err != nil {
+					return err
+				}
+			// if the field is a slice of structs, recurse through them...
+			default:
+				for i := 0; i < field.Len(); i++ {
+					f := field.Index(i)
+					if f.Kind() == reflect.Ptr {
+						f = f.Elem()
+					}
+					if f.Kind() != reflect.Struct {
+						continue
+					}
+					if err := ef.filterField(ctx, f, filterOverrides, opt...); err != nil {
+						return err
+					}
+				}
+			}
+		// if the field is a struct
+		case fkind == reflect.Struct:
+			if err := ef.filterField(ctx, field, filterOverrides, opt...); err != nil {
+				return err
+			}
+
+		case isTaggable:
+			if err := ef.filterTaggable(ctx, taggedInterface, opt...); err != nil {
+				return fmt.Errorf("%s: %w", op, err)
+			}
+		}
+	}
+	return nil
+}
+
+// filterTaggable will filter data that implements the Taggable interface
+func (ef *Filter) filterTaggable(ctx context.Context, t Taggable, _ ...Option) error {
+	const op = "event.(Filter).filterTaggable"
+	if t == nil {
+		return fmt.Errorf("%s: missing taggable interface: %w", op, ErrInvalidParameter)
+	}
+	tags, err := t.Tags()
+	if err != nil {
+		return fmt.Errorf("%s: unable to get tags from taggable interface: %w", op, err)
+	}
+	for _, pt := range tags {
+		value, err := pointerstructure.Get(t, pt.Pointer)
+		if err != nil {
+			if errors.Is(err, pointerstructure.ErrNotFound) {
+				continue
+			} else {
+				return fmt.Errorf("%s: unable to get value using tag pointer structure (pointer == %s): %w", op, pt.Pointer, err)
+			}
+		}
+		rv := reflect.Indirect(reflect.ValueOf(value))
```

Comment:
I think you may need a nil check before this, in case a map has a valid key but `nil` as the value.

Reply:
I believe, filterValue(...) will properly handle it if it's nil.  A call to reflect.Indirect(reflect.ValueOf(nil)) does not panic